AK/HashTable.h

```diff
@@ -410,92 +372,63 @@ class HashTable {
         return false;
     }
 
-    void remove(Iterator iterator)
+    // This invalidates the iterator
+    void remove(Iterator& iterator)
     {
-        VERIFY(iterator.m_bucket);
-        auto& bucket = *iterator.m_bucket;
-        VERIFY(is_used_bucket(bucket.state));
-
-        delete_bucket(bucket);
-        --m_size;
-        ++m_deleted_count;
-
-        rehash_in_place_if_needed();
+        auto* bucket = iterator.m_bucket;
+        VERIFY(bucket);
+        delete_bucket(*bucket);
+        iterator.m_bucket = nullptr;
     }
 
     template<typename TUnaryPredicate>
     bool remove_all_matching(TUnaryPredicate const& predicate)
```

Comment:
I think returning the amount of removed elements is actually more information here.
But that can just be changed when someone does need that information.

Reply:
I agree with you, but I've also tried to not change the API too much. So I'll leave that in as an exercise for the user :^)